tests/NexusMods.UI.Tests/AEndToEndTest.cs

```diff
@@ -0,0 +1,29 @@
+﻿using NexusMods.UI.Tests.Framework;
+
+namespace NexusMods.UI.Tests;
+
+/// <summary>
+/// A base test class for a test that needs the entire app, the MainWindow and a full View Model
+/// </summary>
+public class AEndToEndTest : IAsyncLifetime
+{
+    private readonly AvaloniaApp _app;
+    
+    protected WindowHost? Host { get; private set; }
```

Comment:
Since we're setting this value in `IAsyncLifetime.InitializeAsync`, we can remove nullability from this property:

```csharp
private WindowHost? _host;

// either get fancy with Debug.Assert
public WindowHost Host
{
    get
    {
        Debug.Assert(_host is not null);
        return _host;
    }
}

// or just throw an exception
public WindowHost Host => _host ?? throw new UnreachableException();
```